applications/mne_anonymize/fiffanonymizer.cpp

```diff
@@ -716,282 +227,445 @@ int FiffAnonymizer::censorTag(FiffTag::SPtr outTag,
+
+bool FiffAnonymizer::checkValidFiffFormatVersion() const
+{
+    if(m_pTag->kind == FIFF_FILE_ID)
+    {
+        FIFFLIB::FiffId fileId = m_pTag->toFiffID();
+        int inMayorVersion = (static_cast<uint32_t>(fileId.version) & 0xFFFF0000) >> 16;
+        int inMinorVersion = (static_cast<uint32_t>(fileId.version) & 0x0000FFFF);
+        double inVersion = inMayorVersion + inMinorVersion/10.0;
+
+        if(inVersion > m_dMaxValidFiffVerion) {
+            return false;
+        }
+        return true;
     }
+    return false;
+}
 
-    sizeDiff = outTag->size() - inTag->size();
+//=============================================================================================================
 
-    return sizeDiff;
+int FiffAnonymizer::setFileIn(const QString &sFileIn)
+{
+    QFileInfo fiIn(sFileIn);
+    if(fiIn.exists())
+    {
+        m_fFileIn.setFileName(fiIn.absoluteFilePath());
+        m_bFileInSet = true;
+        return 0;
+    } else {
+        qCritical() << "File doesn't exist.";
+        return 1;
+    }
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setFileIn(const QString &sFileIn)
+QString FiffAnonymizer::getFileNameIn() const
 {
-    m_sFileNameIn = sFileIn;
-    m_fFileIn.setFileName(sFileIn);
+    return m_fFileIn.fileName();
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setFileOut(const QString &sFileOut)
+int FiffAnonymizer::setFileOut(const QString &sFileOut)
 {
-    if(m_fFileIn.fileName().compare(sFileOut,Qt::CaseInsensitive) == 0) {
-        m_bInOutFileNamesEqual = true;
-        QFileInfo outFileInfo(sFileOut);
-        outFileInfo.makeAbsolute();
-        m_sFileNameOut = outFileInfo.absolutePath() + "/" + generateRandomFileName();
+    if(m_bFileInSet)
+    {
+        QFileInfo fiIn(m_fFileIn);
+        QFileInfo fiOut(sFileOut);
+        if(fiOut.absoluteFilePath() == fiIn.absoluteFilePath())
+        {
+            qCritical() << "Both input and output file names are the same.";
+            return 1;
+        } else {
+            m_fFileOut.setFileName(fiOut.absoluteFilePath());
+        }
+        m_bFileOutSet = true;
+        return 0;
     } else {
-        m_bInOutFileNamesEqual = false;
-        m_sFileNameOut = sFileOut;
+        qCritical() << "You need to specify the input file first.";
+        return 1;
     }
+}
+
+//=============================================================================================================
 
-    m_fFileOut.setFileName(m_sFileNameOut);
+QString FiffAnonymizer::getFileNameOut() const
+{
+    return m_fFileOut.fileName();
 }
 
 //=============================================================================================================
 
-void FiffAnonymizer::setVerboseMode(bool bFlag)
+void FiffAnonymizer::setVerboseMode(const bool bFlag)
```

Comment:
`const` is not needed here since the passed value is not a reference

Reply:
I understand, but won't it be of help to the compiler? Again, this is just a stupid trivial program, but I'm trying to learn to do things properly. I won't modify the value, so it will help the compiler (I think).

Reply:
yes it might help the compiler optimize, although this is highly questionable with the function at hand. This discussion might be an interesting read: https://stackoverflow.com/questions/117293/use-of-const-for-function-parameters